MagicalCryptoWallet/Services/IndexBuilderService.cs

```diff
@@ -24,7 +24,67 @@ public class IndexBuilderService
 		private List<FilterModel> Index { get; }
 		private AsyncLock IndexLock { get; }
 
-		private Dictionary<OutPoint, Script> Bech32UtxoSet { get; }
+		private Dictionary<OutPoint, Script> Bech32UtxoSet { get; } //Change to observable concurrent dictionary? https://www.nuget.org/packages/ParallelExtensionsExtras/
```

Comment:
While concurrent collections don't throw exceptions, not all threading issues can be ignored. And they can work in a quite misleading way: https://stackoverflow.com/a/1966462/2061103  

I want it to throw an exception if something is wrong. As you see, right above this line there is an `IndexLock`  for the `Index` list. This is, because the Index may encounter threading issues, but as the code is today `Bech32UtxoSet` won't encounter threading issues, since it's an "internal database" that's being touched in a proper way.

Also performance is quite an issue here, so that's another con against concurrent collections.

Generally I use concurrent collections against lock, whenever
1. I'm lazy.  
2. I'm showing data to the user (that doesn't have to be so accurate.)  

Discussion on this topic can be found here: https://github.com/stratisproject/StratisBitcoinFullNode/issues/617

If you agree, please remove the todo comment here.

Reply:
Sure I will remove. P.S.: I would use the observable part of the dictionary to subcribe the add/remove/clear event to ensure that all changes of the dictionary get noticed.

Reply:
Another way is to implement my own observable dictionary without concurrency: something like this
https://stackoverflow.com/questions/5663395/net-observabledictionary

Reply:
I think it's better as it is now, since it's simpler. Btw, I created my ObservableConcurrentDictionary if you ever need such: https://github.com/nopara73/HiddenWallet/blob/cc8c948ac7a1b9173610145e60c303d71a8c0a66/HiddenWallet/Helpers/ConcurrentObservableDictionary.cs